sycl/test/plugins/level_zero_dynamic_batch_test.cpp

```diff
@@ -0,0 +1,219 @@
+                                          const int m = it[0];
+                                          const int n = it[1];
+                                          Z7[m * N + n] = Y1[m * N + n];
+                                        });
+      });
+      q.submit([&](sycl::handler &h) {
+        h.parallel_for<class u32_copy8>(sycl::range<2>{M, N},
+                                        [=](sycl::id<2> it) {
+                                          const int m = it[0];
+                                          const int n = it[1];
+                                          Z8[m * N + n] = Y1[m * N + n];
+                                        });
+      });
+    }
+    q.wait();
+  }
+
+  validate(Y1, Z1, M * N);
+  validate(Y1, Z2, M * N);
+  validate(Y1, Z3, M * N);
+  validate(Y1, Z4, M * N);
+  validate(Y1, Z5, M * N);
+  validate(Y1, Z6, M * N);
+  validate(Y1, Z7, M * N);
+  validate(Y1, Z8, M * N);
+
+  for (size_t i = 0; i < 5; i++) {
+    q.submit([&](sycl::handler &h) {
+      h.parallel_for<class u32_copy9>(sycl::range<2>{M, N},
+                                      [=](sycl::id<2> it) {
+                                        const int m = it[0];
+                                        const int n = it[1];
+                                        Z1[m * N + n] = Y1[m * N + n];
+                                      });
+    });
+    q.submit([&](sycl::handler &h) {
+      h.parallel_for<class u32_copy10>(sycl::range<2>{M, N},
+                                       [=](sycl::id<2> it) {
+                                         const int m = it[0];
+                                         const int n = it[1];
+                                         Z2[m * N + n] = Y1[m * N + n];
+                                       });
+    });
+    q.submit([&](sycl::handler &h) {
+      h.parallel_for<class u32_copy11>(sycl::range<2>{M, N},
+                                       [=](sycl::id<2> it) {
+                                         const int m = it[0];
+                                         const int n = it[1];
+                                         Z3[m * N + n] = Y1[m * N + n];
+                                       });
+    });
+    q.submit([&](sycl::handler &h) {
+      h.parallel_for<class u32_copy12>(sycl::range<2>{M, N},
+                                       [=](sycl::id<2> it) {
+                                         const int m = it[0];
+                                         const int n = it[1];
+                                         Z4[m * N + n] = Y1[m * N + n];
+                                       });
+    });
+    q.wait();
+  }
+  validate(Y1, Z1, M * N);
+  validate(Y1, Z2, M * N);
+  validate(Y1, Z3, M * N);
+  validate(Y1, Z4, M * N);
+
+  sycl::free(Y1, ctx);
+  sycl::free(Z1, ctx);
+  sycl::free(Z2, ctx);
+  sycl::free(Z3, ctx);
+  sycl::free(Z4, ctx);
+  sycl::free(Z5, ctx);
+  sycl::free(Z6, ctx);
+  sycl::free(Z7, ctx);
+  sycl::free(Z8, ctx);
+
+  return 0;
+}
```

Comment:
I have the feeling this test could have 10% of the current size by keeping the same semantics... Extensive programming vs intensive programming...

Reply:
I agree, it could have been much smaller.  I just copied and pasted to get what I needed for the test.  I'll reduce the test case in a follow-on change.

Reply:
If you run out of imagination :-) https://en.cppreference.com/w/cpp/utility/integer_sequence